utils/ort/src/main/kotlin/CopyrightStatementsProcessor.kt

```diff
@@ -35,6 +35,21 @@ private data class Parts(
     companion object {
         val COMPARATOR = compareBy<Parts>({ it.owner }, { prettyPrintYears(it.years) }, { it.prefix })
     }
+
+    override fun toString() =
+        buildString {
+            append(prefix)
+
+            if (years.isNotEmpty()) {
+                append(" ")
+                append(prettyPrintYears(years))
+            }
+
+            if (owner.isNotEmpty()) {
+                append(" ")
+                append(owner)
```

Comment:
Previously, copyright statements without an owner have been skipped. With this change this is no more the case: it now produces copyright statement without an owner. I prefer skipping statements with empty owners as it was before.

Reply:
> I prefer skipping statements with empty owners as it was before.

It's still like that as the caller still has `if (it.owner.isNotEmpty()) {` like before. The `if (owner.isNotEmpty()) {` introduced here is on top of that. I'll address this temporary "duplication" in a followup PR.

Reply:
My plan is actually to remove the `if (it.owner.isNotEmpty()) {` from the caller and to handle empty owners as unprocessed. I.e. there should be no statement that is neither processed nor unprocessed. And this logic change I'd prefer to keep out of the "factor-out the to-string"- topic.

Reply:
If you're curious, [here's a draft](https://github.com/oss-review-toolkit/ort/pull/6022) that is based on this PR.

Reply:
After looking at this draft, I again ask (as more PRs are necessary unexpectedly) what the goal of this refactoring is, so that I can consider it in the review.

Reply:
I can only repeat myself from [here](https://github.com/oss-review-toolkit/ort/pull/6014#discussion_r1009234410). There still is no ultimate goal or hidden agenda. It's more like an unplanned "code audit" that was partly triggered by your review comments actually. And while playing around with what you suggested (like changing the return value of `process` instead of making `determineParts` public), I discovered issues like statements without owners being swallowed, and the returned owner being dependent on the order of the original statements. I fixed these issues now as I discovered them, but I have no further concrete plans for this class (unless more issues pop up).

Reply:
Thanks! 

> I can only repeat myself from 

That comment says "this PR concludes that work", so the situation has changed which is why I think asking the question again is valid.

> I discovered issues like statements without owners being swallowed

To me it is not obvious whether this is an issue. Let's just do a quick poll on this from the core maintainers.

Reply:
> That comment says "this PR concludes that work", so the situation has changed which is why I think asking the question again is valid.

Sure, asking does not hurt. But I also wrote "for the time being" there, as I of course couldn't rule out more work coming up due to the reviews.

> To me it is not obvious whether this is an issue. Let's just do a quick poll on this from the core maintainers.

Fine with me. But can we proceed with *this* PR then?

Reply:
> Fine with me. But can we proceed with _this_ PR then?

Not sure if I'm missing something but doesn't this PR assume that swallowing will be removed? e.g. the code change in this PR would be different otherwise?

Reply:
> e.g. the code change in this PR would be different otherwise?

No, it wouldn't. I wrote this code before I had the idea to record empty owners as unprocessed statements.